apps/callscreen/js/handled_call.js

```diff
@@ -289,65 +265,70 @@ HandledCall.prototype.updateCallNumber = function hc_updateCallNumber() {
     }
     if (contact) {
       var primaryInfo = Utils.getPhoneNumberPrimaryInfo(matchingTel, contact);
-      var contactCopy = {
-        id: contact.id,
-        name: contact.name,
-        org: contact.org,
-        tel: contact.tel
-      };
       if (primaryInfo) {
-        node.textContent = primaryInfo;
-        self._cachedInfo = primaryInfo;
+        self.replacePhoneNumber({ raw: primaryInfo[0] }, 'end');
```

Comment:
The primaryInfo is basically the name field of the associated MozContact, this field is an array and we're taking the first entry by default: https://developer.mozilla.org/en-US/docs/Web/API/MozContact/name#Value